src/shogun/lib/parameter_observers/ParameterObserverCV.cpp

```diff
@@ -57,27 +56,14 @@ CParameterObserverCV::~CParameterObserverCV()
 
 void CParameterObserverCV::on_next(const shogun::TimedObservedValue& value)
 {
-	CHECK_OBSERVED_VALUE_TYPE(value.first.get_type());
+	CrossValidationStorage* recalled_value = value.first->get<CrossValidationStorage*>("value");
+	SG_REF(recalled_value);
 
-	if (value.first.get_value().type_info().hash_code() ==
-	    typeid(CrossValidationStorage*).hash_code())
-	{
-		CrossValidationStorage* recalled_value =
-		    any_cast<CrossValidationStorage*>(value.first.get_value());
-		SG_REF(recalled_value);
-
-		/* Print information on screen if enabled*/
-		if (m_verbose)
-			print_observed_value(recalled_value);
+	/* Print information on screen if enabled*/
+	if (m_verbose)
```

Comment:
I think printing on screen is generally a bad thing to do. Rather print to a stream or shogun io or something....

Reply:
SG_PRINT is fine? It should be okay since it uses the `SGIO` object which could be defined by the user.